CoursesManager.UI/Repositories/RegistrationRepository/RegistrationRepository.cs

```diff
@@ -48,52 +48,52 @@ public List<Registration> RefreshAll()
         }
     }
 
-    public void Add(Registration data)
+    public void Add(Registration registration)
     {
         lock (_allRegistrations)
         {
-            ArgumentNullException.ThrowIfNull(data);
+            ArgumentNullException.ThrowIfNull(registration);
 
-            _registrationDataAccess.Add(data);
-            _allRegistrations.Add(data);
+            _registrationDataAccess.Add(registration);
+            _allRegistrations.Add(registration);
         }
     }
 
-    public void Update(Registration data)
+    public void Update(Registration registration)
     {
         lock (_allRegistrations)
         {
-            ArgumentNullException.ThrowIfNull(data);
+            ArgumentNullException.ThrowIfNull(registration);
 
-            _registrationDataAccess.Update(data);
+            _registrationDataAccess.Update(registration);
 
-            if (!_allRegistrations.Contains(data))
+            if (!_allRegistrations.Contains(registration))
```

Comment:
Misschien kun je dit voor safety beter doen op basis van registration.id want is niet per se gegarandeerd dat de registration die je meegeeft er niet in zit als je contains doet. Contains checkt reference equals en niet data equals

Reply:
Als ik wijzig naar registration.Id krijg ik foutmelding dat registration.Id not null is

Reply:
Lees goed ik zeg Not Null en daar gaat die wijziging daar niet mee overweg :) specifiek convert in van registration of zoiets probeer maar :)